app/livechat/client/voip.ts

```diff
@@ -4,9 +4,7 @@ import { MessageTypes } from '../../ui-utils/client';
 import { IVoipMessage } from '../../../definition/IMessage';
 
 type IMessageFuncReturn = {
-	at?: string;
-	duration?: string;
-	time?: string;
+	[k: string]: string | undefined;
```

Comment:
Here you're doing `type IMessageFuncReturn = Record<string, string | undefined>`, which is not good. I believe `IMessageFuncReturn` should be a union.

```ts
type IMessageFuncReturn =
  | {
    at: string | undefined;
    time: string;
  }
  | {
    duration: string;
  }
  | {
    at: string;
  }
  | {
    comment: string;
  }
```

Not ideal, but at least type-safe.

Reply:
@KevLehman ?

Reply:
Can you check again? I changed this to a union (not my personal favorite but works) and added some types to the MessageTypes class.